sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,560 @@
+        preds = np.array(
+            [
+                self.classes_[int(rng.choice(np.flatnonzero(prob == prob.max())))]
+                for prob in probas
+            ]
+        )
+
+        return (preds, out[1], out[2]) if self.return_safety_decisions else preds
+
+    def predict_proba(self, X, state_info=None):
+        """Decide on the safety of an early classification.
+
+        Parameters
+        ----------
+        X : 3D np.array (any number of dimensions, equal length series)
+                of shape [n_instances, n_dimensions, series_length]
+            or 2D np.array (univariate, equal length series)
+                of shape [n_instances, series_length]
+            or pd.DataFrame with each column a dimension, each cell a pd.Series
+                (any number of dimensions, equal or unequal length series)
+            or of any other supported Panel mtype
+                for list of mtypes, see datatypes.SCITYPE_REGISTER
+                for specifications, see examples/AA_datatypes_and_datasets.ipynb
+        state_info : List or None
+            A List containing the state info for each decision in X. contains
+            information for future decisions on the data. Inputs should be None or an
+            empty List for the first decision made, the returned List new_state_info for
+            subsequent decisions.
+            If no state_info is provided and the input series_length is greater than
+            the first classification_points time stamp, all previous time stamps are
+            considered up to the input series_length and the probabilities for the first
+            safe prediction are returned.
+
+        Returns
+        -------
+        y : 2D array of shape [n_instances, n_classes] - predicted class probabilities
+            1st dimension indices correspond to instance indices in X
+            2nd dimension indices correspond to possible labels (integers)
+            (i, j)-th entry is predictive probability that i-th instance is of class j
+        decisions : 1D bool array
+            An array of booleans, containing the decision of whether a prediction is
+            safe to use or not. Returned if return_safety_decisions is True.
+        new_state_info : 2D int array
+            An array containing the state info for each decision in X, contains
+            information for future decisions on the data. Returned if
+            return_safety_decisions is True.
+        """
+        self.check_is_fitted()
+
+        # boilerplate input checks for predict-like methods
+        X = self._check_convert_X_for_predict(X)
+
+        return self._predict_proba(X, state_info=state_info)
+
+    def _predict_proba(self, X, state_info=None):
+        n_instances, _, series_length = X.shape
+
+        # maybe use the largest index that is smaller than the series length
+        next_idx = self._get_next_idx(series_length)
+
+        if next_idx == -1:
```

Comment:
I'm unsure why this is required instead of the dictionary. Is next_idx supposed to mean the index of the timestamp after the current input?

I don't think it can be -1 also.

Reply:
I think just go with the full series length. self.classification_points is not required as an input, it defaults to Nones which gives <=20 values depending on the input series length.

self._classification_points removes any duplicates, any time stamps below a minimum length and adds the final series length, so they may not match up well either.